test/OpenIddict.Server.Tests/OpenIddictServerExtensionsTests.cs

```diff
@@ -0,0 +1,403 @@
+        //                                         service.ImplementationType == typeof(OpenIddictServerEventDispatcher));
+        //}
+
+        //[Fact]
+        //public void AddServer_RegistersHandler()
+        //{
+        //    // Arrange
+        //    var services = new ServiceCollection();
+        //    var builder = new OpenIddictBuilder(services);
+
+        //    // Act
+        //    builder.AddServer();
+
+        //    // Assert
+        //    Assert.Contains(services, service => service.Lifetime == ServiceLifetime.Scoped &&
+        //                                         service.ServiceType == typeof(OpenIddictServerHandler) &&
+        //                                         service.ImplementationType == typeof(OpenIddictServerHandler));
+        //}
+
+        //[Fact]
+        //public void AddServer_RegistersProvider()
+        //{
+        //    // Arrange
+        //    var services = new ServiceCollection();
+        //    var builder = new OpenIddictBuilder(services);
+
+        //    // Act
+        //    builder.AddServer();
+
+        //    // Assert
+        //    Assert.Contains(services, service => service.Lifetime == ServiceLifetime.Scoped &&
+        //                                         service.ServiceType == typeof(OpenIddictServerProvider) &&
+        //                                         service.ImplementationFactory != null);
+        //}
+
+        [Fact]
+        public void AddServer_ResolvingProviderThrowsAnExceptionWhenCoreServicesAreNotRegistered()
+        {
+            // Arrange
+            var services = new ServiceCollection();
+            var builder = new OpenIddictBuilder(services);
+
+            // Act
+            builder.AddServer();
+
+            // Assert
+            var provider = services.BuildServiceProvider();
+
+            var exception = Assert.Throws<InvalidOperationException>(() => provider.GetRequiredService<OpenIddictServerConfiguration>());
+
+            Assert.NotNull(exception);
+        }
+
+        [Theory]
+        [InlineData(typeof(IPostConfigureOptions<OpenIddictServerOptions>), typeof(OpenIddictServerConfiguration))]
+        //[InlineData(typeof(IPostConfigureOptions<OpenIddictServerOptions>), typeof(OpenIdConnectServerInitializer))]
+        public void AddServer_RegistersConfiguration(Type serviceType, Type implementationType)
+        {
+            // Arrange
+            var services = new ServiceCollection();
+            var builder = new OpenIddictBuilder(services);
+
+            // Act
+            builder.AddServer();
+
+            // Assert
+            Assert.Contains(services, service => service.ServiceType == serviceType &&
+                                                 service.ImplementationType == implementationType);
+        }
+
+        //[Fact]
+        //public void AddServer_RegistersAuthenticationScheme()
```

Comment:
It's no longer something the generic server services or the ASP.NET Core host do in 3.0: the `OpenIddictServerAspNetCoreHandler` is now lazily registered by `OpenIddictServerAspNetCoreConfiguration`.

Reply:
Alright - I have removed that commented out test. There are a few more that I was not sure about - are the remaining commented out tests also able to be removed completely?

The list of tests that are still commented out is:
- `AddServer_RegistersCachingServices()`
- `AddServer_RegistersEventService()`
- `AddServer_RegistersHandler()`
- `AddServer_RegistersProvider()`
- `AddServer_ThrowsAnExceptionWhenSchemeIsAlreadyRegisteredWithDifferentHandlerType()`

Reply:
> `AddServer_RegistersEventService()`
> `AddServer_RegistersProvider()`

The event service was replaced by `IOpenIddictServerDispatcher` and `IOpenIddictServerFactory`.

> `AddServer_RegistersHandler()`
> `AddServer_ThrowsAnExceptionWhenSchemeIsAlreadyRegisteredWithDifferentHandlerType()`

The authentication handler is now registered by the ASP.NET Core host in `OpenIddictServerAspNetCoreConfiguration`, so you can remove these tests from `OpenIddictServerExtensionsTests`.